src/app/dataset-view/dataset.service.ts

```diff
@@ -49,19 +50,24 @@ export class AppDatasetService {
     }
 
     public requestDatasetMainData(info: DatasetInfo): void {
-        this.datasetApi
-            .getDatasetMainData(info)
-            .subscribe((data: GetDatasetMainDataQuery) => {
-                if (isNil(data.datasets.byOwnerAndName)) {
-                    throw new Error("Dataset not resolved by ID");
+        this.datasetApi.getDatasetMainData(info).subscribe(
+            (data: GetDatasetMainDataQuery) => {
+                if (data.datasets.byOwnerAndName) {
+                    this.datasetUpdate(data.datasets.byOwnerAndName);
+                    this.overviewTabDataUpdate(data);
+                    this.dataTabDataUpdate(data);
+                    this.metadataTabDataUpdate(data);
+                    this.lineageTabDataUpdate(data);
+                } else {
+                    this.errorService.processError(
+                        new Error(ErrorTexts.ERROR_DATASET_NOT_FOUND),
```

Comment:
Please also report a name of the dataset.
I think handling this particular error (not every) should also redirect to main page, because components are left in incompletely initialized state.